rs-react-app/src/components/contentBlock/contentBlock.tsx

```diff
@@ -0,0 +1,249 @@
+  }, [apiPaginaion.parameter])
+
+  function ErrorClick () {
+    setContentState(prev => ({
+      ...prev, 
+      errorMessage: true
+    }))
+  }
+
+  function closeDetailView() {
+    localStorage.clear()
+    setContentState(prev => ({
+      ...prev,
+      loadingDetails: false,
+      searchResult: '',
+    }));
+    onPaginationClick('')
+  }
+
+  function masterDetail (berryName: string) {
+    onPaginationClick(berryName)
+  }
+  usePaginationHook(param, setContentState, searchParams, updateSearchParams)
+
+  function onPaginationClick (param: number | string) {
+    updateParam(param)
+  }
+
+  function paginationControl (param: number) {
+    setContentState(prev => ({
+      ...prev, 
+      loading: true, 
+    }))
+    updateApiPagination(({
+      loading: true, 
+      parameter: param
+    }))
+    setTimeout(() => {
+      setContentState(prev => ({
+        ...prev, 
+        loading: false, 
+      }))
+    }, 2000)
+    updateParam(param)
+    // const newParams = new URLSearchParams(searchParams);
+    // newParams.set('q', String(param));
+    // updateSearchParams(newParams);
+    console.log('paginationControl' + param)
+  }
+
+  function checkboxControl(item: string, checked: boolean) {
+
+    if(checked) {
+      fetch(`https://pokeapi.co/api/v2/berry/${item}/`)
+        .then(response => {
+          return response.json()
+        })
+        .then(result => {
+          dispatch(addData(result))
+          dispatch(addEl(item))
+          console.log(result)
+        })
+    }
+    else {
+      dispatch(removeEl(item))
+      dispatch(removeSpecificData(item))
+    }
+  }
+
+  if(contentState.errorMessage) {
+    throw new Error("I am an artificial error!");
+  }
+
+  if(contentState.loading) {
+    return (
+      <section className='contentBlock' data-testid="content-block">
+        <Preloader testId='loader-icon'></Preloader>
+      </section>
+    )
+  }
+
+  return (
+    <section className='contentBlock' data-testid="content-block">
+      <div className='contentBlock__top'>
+        <HandleForm onClick={onPaginationClick}></HandleForm>
+      </div>
+      <div className='contentBlock__middle'>
+        <ul className='listApi'>
+          {contentState.data && contentState.data.map((berry) => (
+            <li data-testid='search-el' key={berry.name} onClick={() => masterDetail(berry.name)} className='listApi__el'>
+              <h2 >{berry.name} - </h2>
+              <div className='listApi__el-info'>
+                <p >url:</p>
+                <p >{berry.url}</p>
+              </div>
+              <label>
+                <input
+                  data-testid="checkbox-test"
+                  type="checkbox"
+                  checked={selectedElements.includes(berry.name)}
+                  onChange={e => {
+                    // e.stopPropagation()
+                    console.log('on change checkbox')
+                    checkboxControl(berry.name, e.target.checked)
+                  }}
```

Comment:
- cleanup please comments
- same here as for logs

```suggestion
                  onChange={e => checkboxControl(berry.name, e.target.checked) }
```